### PR TITLE
Configure LND connection as part of onboarding process

### PR DESCRIPTION
**Pull request by mrfelton**
This is a new PR that PR supersedes the one in https://github.com/LN-Zap/zap-desktop/pull/327 to address #326 and #167 (supporting custom lnd connectionbs)

In this version, you can configure your lighting node connection as part of the onboarding process where you can choose to run in 'local' mode (which works the as before) or 'custom' mode where you can specify a hostname and path to tls cert and macaroon files.

Settings are persisted to and read from a settings file using `electron-store`.

### CODE REVIEW

File: app/components/Onboarding/ConnectionDetails.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        value={connectionCert}
+        onChange={event => setConnectionCert(event.target.value)}
+      />
+    </div>
+    <div>
+      <label htmlFor='connectionMacaroon'>Macaroon:</label>
+      <input
+        type='text'
+        id='connectionMacaroon'
+        placeholder='Math to the lnd macaroon file'
```
Comment (such) [inline app/components/Onboarding/ConnectionDetails.js]: typo: Path

Comment (fullcubeci) [reply app/components/Onboarding/ConnectionDetails.js]: Thanks. Fixed.

**Comment by mariodian** [comment]
tACK.

For those wanting to try:

1) add `rpclisten=0.0.0.0:10009` to lnd.conf
2) copy `tls.cert` and `admin.macaroon` to your local machine
3) open the port on the server `sudo iptables -A INPUT -p tcp --dport 10009 -j ACCEPT`
4) restart lnd
5) run Zap and point it to `your.server.ip.address:10009` and cert and macaroon files (absolute path)

screen shot 2018-04-06 at 6 43 32 pm

Thanks!

File: app/components/Onboarding/Onboarding.js [hunk 2]
```diff
@@ -38,12 +46,43 @@ const Onboarding = ({
 }) => {
   const renderStep = () => {
     switch (step) {
+      case 'connection.type':
+        return (
+          <FormContainer
+            title='1. Ligntning Connection'
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: we can remove the numbers from the title here, they aren't used in other steps

File: app/components/Onboarding/Onboarding.js [hunk 3]
```diff
@@ -38,12 +46,43 @@ const Onboarding = ({
...
+            next={() => changeStep(connectionType === 'local' ? 1 : 'connection.details')}
+          >
+            <ConnectionType {...connectionTypeProps} />
+          </FormContainer>
+        )
+
+      case 'connection.details':
+        return (
+          <FormContainer
+            title='2. Connection details'
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: can remove number here as well

File: app/components/Onboarding/Onboarding.js [hunk 4]
```diff
@@ -38,12 +46,43 @@ const Onboarding = ({
 }) => {
   const renderStep = () => {
     switch (step) {
+      case 'connection.type':
+        return (
+          <FormContainer
+            title='1. Ligntning Connection'
+            description='How will you connect to the Lightning Network?'
+            back={null}
+            next={() => changeStep(connectionType === 'local' ? 1 : 'connection.details')}
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: hmmmmm. i don't like the fact that `steps` are now being stored as a string or a number. strings may be cleaner, but we should pick one. i'd say change `connection.details` and `connection.type` to numbers for now and we can get a separate PR do convert to strings after this is merged

File: app/components/Onboarding/Onboarding.js [hunk 5]
```diff
@@ -53,9 +92,9 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Autopilot'
-            description='Autopilot is an automatic network manager. Instead of manually adding people to build your network to make payments, enable autopilot to automatically connect you to the Lightning Network using 60% of your balance.' // eslint-disable-line
+            description="Autopilot is an automatic network manager. Instead of manually adding people to build your network to make payments, enable autopilot to automatically connect you to the Lightning Network using 60% of your balance." // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: we use single quotes

File: app/components/Onboarding/Onboarding.js [hunk 6]
```diff
@@ -64,7 +103,7 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Welcome back!'
-            description='Enter your wallet password or create a new wallet' // eslint-disable-line
+            description="Enter your wallet password or create a new wallet" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single quotes

File: app/components/Onboarding/Onboarding.js [hunk 7]
```diff
@@ -75,11 +114,13 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Welcome!'
-            description='Looks like you are new here. Set a password to encrypt your wallet. This password will be needed to unlock Zap in the future' // eslint-disable-line
+            description="Looks like you are new here. Set a password to encrypt your wallet. This password will be needed to unlock Zap in the future" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single quotes

File: app/components/Onboarding/Onboarding.js [hunk 8]
```diff
@@ -90,8 +131,8 @@ const Onboarding = ({
       case 5:
         return (
           <FormContainer
-            title={'Alright, let\'s get set up'}
-            description='Would you like to create a new wallet or import an existing one?' // eslint-disable-line
+            title={"Alright, let's get set up"}
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single

File: app/components/Onboarding/Onboarding.js [hunk 9]
```diff
@@ -90,8 +131,8 @@ const Onboarding = ({
...
+            description="Would you like to create a new wallet or import an existing one?" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single

File: app/components/Onboarding/Onboarding.js [hunk 10]
```diff
@@ -102,7 +143,7 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Save your wallet seed'
-            description='Please save these 24 words securely! This will allow you to recover your wallet in the future' // eslint-disable-line
+            description="Please save these 24 words securely! This will allow you to recover your wallet in the future" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single

File: app/components/Onboarding/Onboarding.js [hunk 11]
```diff
@@ -113,11 +154,13 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Re-enter your seed'
-            description='Yeah I know, might be annoying, but just to be safe!' // eslint-disable-line
+            description="Yeah I know, might be annoying, but just to be safe!" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single

File: app/components/Onboarding/Onboarding.js [hunk 12]
```diff
@@ -129,11 +172,13 @@ const Onboarding = ({
         return (
           <FormContainer
             title='Encrypt your seed'
-            description='Totally optional, but we encourage it. Set a password that will be used to encrypt your wallet seed' // eslint-disable-line
+            description="Totally optional, but we encourage it. Set a password that will be used to encrypt your wallet seed" // eslint-disable-line
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: single

File: app/components/Onboarding/Onboarding.js [hunk 13]
```diff
@@ -129,11 +172,13 @@ const Onboarding = ({
...
             back={() => changeStep(6)}
             next={() => {
               // dont allow the user to move on if the confirmation password doesnt match the original password
-              if (newAezeedPasswordProps.showAezeedPasswordConfirmationError) { return }
+              if (newAezeedPasswordProps.showAezeedPasswordConfirmationError) {
```
Comment (JimmyMow) [inline app/components/Onboarding/Onboarding.js]: are you breaking these conditionals into multiple lines for any particular reason?

**Comment by JimmyMow** [comment]
@mrfelton i just skimmed this and gave some syntax feedback. will clone this down tomorrow and take a deeper look. 

great work! this is a popular PR, let's get this merged soon :)

File: app/components/Onboarding/Onboarding.js [hunk 14]
```diff
@@ -38,12 +46,43 @@ const Onboarding = ({
 }) => {
   const renderStep = () => {
     switch (step) {
+      case 0.1:
+        return (
+          <FormContainer
+            title='Ligntning Connection'
```
Comment (LaurensN) [inline app/components/Onboarding/Onboarding.js]: Small typo here: `Ligntning -> Lightning`

Comment (mrfelton) [reply app/components/Onboarding/Onboarding.js]: Thanks @LaurensN - fixed.

**Comment by knutnyg** [comment]
Did this PR die? 💀

**Comment by Bladez1753** [comment]
https://github.com/LN-Zap/zap-desktop/pull/351#issuecomment-379513472

**Comment by Bladez1753** [comment]
> [#351 (comment)](https://github.com/LN-Zap/zap-desktop/pull/351#issuecomment-379513472)

https://github.com/LN-Zap/zap-desktop/pull/351#issuecomment-1236301547

**Pull request merged**